backend/rdf4j/src/test/java/org/dotwebstack/framework/backend/rdf4j/query/SubjectQueryBuilderTest.java

```diff
@@ -0,0 +1,200 @@
+            .silent(false)
+            .strict(true)
+            .create();
+
+  @Mock
+  private QueryEnvironment environmentMock = mock(QueryEnvironment.class);
+
+  @Mock
+  private NodeShapeRegistry registryMock = mock(NodeShapeRegistry.class);
+
+  @Mock
+  private NodeShape nodeShapeMock = mock(NodeShape.class);
+
+  @Mock
+  GraphQLObjectType objectTypeMock = mock(GraphQLObjectType.class);
+
+  private SubjectQueryBuilder subjectQueryBuilder;
+
+  @BeforeEach
+  void setUp() {
+    when(environmentMock.getNodeShapeRegistry()).thenReturn(registryMock);
+    when(environmentMock.getObjectType()).thenReturn(objectTypeMock);
+    when(environmentMock.getNodeShapeRegistry().get(any(GraphQLObjectType.class)))
+            .thenReturn(nodeShapeMock);
+    subjectQueryBuilder = SubjectQueryBuilder.create(environmentMock, jexlEngine);
+  }
+
+  private GraphQLDirective getValidSparqlDirective() {
+    return GraphQLDirective
+            .newDirective()
+            .name("sparql")
+            .argument(GraphQLArgument
+                    .newArgument()
+                    .name("offset")
+                    .type(Scalars.GraphQLString)
+                    .value("(page - 1) * pageSize")
+                    .build())
+            .argument(GraphQLArgument
+                    .newArgument()
+                    .name("limit")
+                    .type(Scalars.GraphQLString)
+                    .value("pageSize")
+                    .build())
+            .build();
+  }
+
+  private GraphQLDirective getInvalidSparqlDirective() {
+    return GraphQLDirective
+            .newDirective()
+            .name("sparql")
+            .argument(GraphQLArgument
+                    .newArgument()
+                    .name("offset")
+                    .type(Scalars.GraphQLString)
+                    .value("this is an invalid expression")
+                    .build())
+            .argument(GraphQLArgument
+                    .newArgument()
+                    .name("limit")
+                    .type(Scalars.GraphQLString)
+                    .value("this is an invalid expression")
+                    .build())
+            .build();
+  }
+
+  @Test
+  void test_pagingParameters_whenNoValuesSet() {
+    GraphQLDirective validSparqlDirective = getValidSparqlDirective();
+    Map<String, Object> arguments = new HashMap<>();
+
+    MapContext context = new MapContext(arguments);
+
+    assertThrows(org.apache.commons.jexl3.JexlException.Variable.class, () ->
+            subjectQueryBuilder.getLimitFromContext(context, validSparqlDirective));
+    assertThrows(org.apache.commons.jexl3.JexlException.Variable.class, () ->
+            subjectQueryBuilder.getOffsetFromContext(context, validSparqlDirective));
+  }
+
+  @Test
+  void test_pagingParameters_withValidPageAndPageSize() {
+    GraphQLDirective validSparqlDirective = getValidSparqlDirective();
+    Map<String, Object> arguments = new HashMap<>();
+    arguments.putIfAbsent("page", 5);
```

Comment:
gewoon dit?:
```suggestion
    arguments.put("page", 5);
```

Dit kan ook:
```suggestion
    Map<String, Object> arguments = Map.of("page", 5, "pageSize", 12);
```